FirebaseInstallations/Source/Library/InstallationsIDController/FIRInstallationsIDController.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class FBLPromise<ValueType>;
+@class FIRInstallationsItem;
+
+/**
+ * The class is responsible for managing FID for a given `FIRApp`.
+ */
+@interface FIRInstallationsIDController : NSObject
+
+- (instancetype)initWithGoogleAppID:(NSString *)appID appName:(NSString *)appName;
+
+- (FBLPromise<FIRInstallationsItem *> *)getInstallationItem;
```

Comment:
I don't see it gets called from FIRInstallations so I wonder if we should have a test app that triggers the public API which calls all the new methods you created. Otherwise, it's hard to vision what this new created method for without seeing it called any where from the public API chain. I think it might help us to have a end-to-end flow during development.

Reply:
It is called from [here](https://github.com/firebase/firebase-ios-sdk/blob/640751e49f03eb30408d1bef9cf9731c78104330/FirebaseInstallations/Source/Library/FIRInstallations.m#L111)

A test app is in my TODO list. I assume it will be used for manual and automated testing. I think to add it when there is something to test end-to-end.

Reply:
aha, right it's moved from another class. Got it